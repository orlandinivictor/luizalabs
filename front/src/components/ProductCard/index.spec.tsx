import { describe, expect, it, vi } from "vitest";
import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import { ProductCard } from ".";

const productMock = {
  code: "3R2-0087-240-02",
  name: "Kit 4 Camiseta Masculina Segunda Pele Proteção Solar UV Térmica Praia Manga Longa Blusa Camisa - Cinza+Azul",
  available: false,
  visible: true,
  details: {
    name: "Kit 4 Camiseta Masculina Segunda Pele Proteção Solar UV Térmica Praia Manga Longa Blusa Camisa - Cinza+Azul",
    description:
      "As camisetas proteção UV masculina da Dellas, são produzidas em 100% Poliéster, permitindo movimentos mais intensos, e por um maior tempo de forma confortável. Elas se ajustam perfeitamente ao corpo, podendo ser usadas como segunda pele.",
  },
  priceInCents: "20000",
  salePriceInCents: "8690",
  rating: 3.5,
  image:
    "https://static.netshoes.com.br/produtos/kit-4-camiseta-masculina-segunda-pele-protecao-solar-uv-termica-praia-manga-longa-blusa-camisa/40/3R2-0087-240/3R2-0087-240_zoom1.jpg?ts=1690411979?ims=544x",
  stockAvailable: false,
};

describe("ProductCard Tests", () => {
  it("match snapshot", () => {
    const domTree = renderer
      .create(<ProductCard product={productMock} />)
      .toJSON();

    expect(domTree).toMatchSnapshot();
  });

  it("match snapshot with 4+ stars", () => {
    const domTree = renderer
      .create(<ProductCard product={{ ...productMock, rating: 4.5 }} />)
      .toJSON();

    expect(domTree).toMatchSnapshot();
  });

  it("should be update wishlist", () => {
    const setWishlist = vi.fn();

    render(
      <ProductCard
        product={productMock}
        setWishlist={setWishlist}
        pathname="wishlist"
      />
    );

    fireEvent.click(screen.getByRole("button"));
    expect(setWishlist).toBeCalled();

    fireEvent.click(screen.getByRole("button"));
    expect(setWishlist).toBeCalledWith([]);
  });
});
