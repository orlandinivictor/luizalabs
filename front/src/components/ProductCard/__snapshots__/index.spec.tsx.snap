// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard Tests > match snapshot 1`] = `
<div
  className="sc-braxZu dOGExz"
>
  <button
    className="sc-gJhJTp hESnrE"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
      />
    </svg>
  </button>
  <img
    alt="Kit 4 Camiseta Masculina Segunda Pele Proteção Solar UV Térmica Praia Manga Longa Blusa Camisa - Cinza+Azul"
    src="https://static.netshoes.com.br/produtos/kit-4-camiseta-masculina-segunda-pele-protecao-solar-uv-termica-praia-manga-longa-blusa-camisa/40/3R2-0087-240/3R2-0087-240_zoom1.jpg?ts=1690411979?ims=544x"
  />
  <h2>
    Kit 4 Camiseta Masculina Segunda Pele Proteção Solar UV Térmica Praia Manga Longa Blusa Camisa - Cinza+Azul
  </h2>
  <div
    className="pricing"
  >
    <div
      className="rating"
    >
      <div
        className="sc-elDIKY kPXcsD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY kPXcsD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY kPXcsD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY ejUDrD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY iWvPbn"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <span>
        3.5
      </span>
    </div>
    <span
      className="originalPrice"
    >
      R$ 200,00
    </span>
    <span
      className="finalPrice"
    >
      R$ 86,90
    </span>
  </div>
</div>
`;

exports[`ProductCard Tests > match snapshot with 4+ stars 1`] = `
<div
  className="sc-braxZu dOGExz"
>
  <button
    className="sc-gJhJTp hESnrE"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
      />
    </svg>
  </button>
  <img
    alt="Kit 4 Camiseta Masculina Segunda Pele Proteção Solar UV Térmica Praia Manga Longa Blusa Camisa - Cinza+Azul"
    src="https://static.netshoes.com.br/produtos/kit-4-camiseta-masculina-segunda-pele-protecao-solar-uv-termica-praia-manga-longa-blusa-camisa/40/3R2-0087-240/3R2-0087-240_zoom1.jpg?ts=1690411979?ims=544x"
  />
  <h2>
    Kit 4 Camiseta Masculina Segunda Pele Proteção Solar UV Térmica Praia Manga Longa Blusa Camisa - Cinza+Azul
  </h2>
  <div
    className="pricing"
  >
    <div
      className="rating"
    >
      <div
        className="sc-elDIKY kPXcsD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY kPXcsD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY kPXcsD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY kPXcsD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <div
        className="sc-elDIKY ejUDrD"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30.335938 12.546875 L 20.164063 11.472656 L 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 L 24.859375 29.398438 L 22.738281 19.394531 Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 16 2.132813 L 11.835938 11.472656 L 1.664063 12.546875 L 9.261719 19.394531 L 7.140625 29.398438 L 16 24.289063 Z"
          />
        </svg>
      </div>
      <span>
        4.5
      </span>
    </div>
    <span
      className="originalPrice"
    >
      R$ 200,00
    </span>
    <span
      className="finalPrice"
    >
      R$ 86,90
    </span>
  </div>
</div>
`;
